aries-site/src/layouts/navigation/Search.js

```diff
@@ -131,22 +173,49 @@ export const Search = ({ setOpen }) => {
   const onEnter = () => {
     if (value) {
       if (suggestions.length === 1) {
-        router.push(nameToPath(suggestions[0].value.title));
+        router.push({
+          pathname: nameToPath(suggestions[0].value.title),
+          query: `q=${value}`,
+        });
+        onClose();
       } else {
+        // TODO: Revisit this case and matching scenario
         const matches = allSuggestions.filter(
           c => c.label.toLowerCase() === value.toLowerCase(),
         );
         if (matches.length === 1) {
-          router.push(nameToPath(matches[0].value.title));
+          router.push({
+            pathname: nameToPath(matches[0].value.title),
+            query: `q=${value}`,
+          });
+          onClose();
         }
       }
     }
   };
 
   const onSelect = event => {
-    const href = nameToPath(event.item.value.title);
+    const { matches, title } = event.item.value;
+    const href = nameToPath(title);
+
+    let id;
+    if (matches && matches.length > 0) {
+      const { index, input } = matches[0];
+      const regexp = new RegExp(/#{1,} (...+?)(?<= ~{2})/, 'g');
```

Comment:
Firefox doesn't support 'lookback' in regular expressions. I think this is why the design system site isn't working in safari.

https://caniuse.com/js-regexp-lookbehind
https://stackoverflow.com/questions/51568821/works-in-chrome-but-breaks-in-safari-invalid-regular-expression-invalid-group/51568859

Reply:
Opened a new issue for this: https://github.com/grommet/hpe-design-system/issues/2053